### PR TITLE
Add white balance node for the basler camera

### PR DESCRIPTION
**Pull request by Flova**
This pull request adds an cpp node that adjusts the white balance of the incoming image.
Note: This is only the node implementation, I build a nodelet version later on to improve the performance.
After taking to Timon and Jonas, I also switched from an ROI fcnn debug output to an Image message.

### CODE REVIEW

**Comment by timonegk** [comment]
I cleaned up a little. The FCNN output publishing will be moved to a separate pull request, this is for white balancing only.

File: white_balancer/src/white_balancer.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+void WhiteBalancer::set_temp(int temp)
+{
+    // Round temp
+    WhiteBalancer::temp = 100 * ((int) temp / 100);
+}
+
+void WhiteBalancer::imageCallback(const sensor_msgs::ImageConstPtr& msg)
+{
+     // Get starting timepoint 
+    auto start = std::chrono::high_resolution_clock::now();
```
Comment (timonegk) [inline white_balancer/src/white_balancer.cpp]: The timing is not really necessary. (See also lines 190-193 and 9-10)

Comment (Flova) [reply white_balancer/src/white_balancer.cpp]: yea this was debugging stuff, thanks for reminding.

File: white_balancer/package.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>white_balancer</name>
+  <version>0.0.0</version>
+  <description>The white_balanceing package</description>
+
+  <maintainer email="7vahl@informatik.uni-hamurg.de">Flo^2</maintainer>
```
Comment (timonegk) [inline white_balancer/package.xml]:
```suggestion
  <maintainer email="7vahl@informatik.uni-hamurg.de">Florian Vahl</maintainer>
```

File: white_balancer/package.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  <depend>image_transport</depend>
+  <depend>roscpp</depend>
+  <depend>sensor_msgs</depend>
+  <depend>std_msgs</depend>
+  <depend>OpenCV</depend>
+
+  <export>
+    <bitbots_documentation>
+      <language>c++</language>
+      <status>unknown</status>
```
Comment (timonegk) [inline white_balancer/package.xml]:
```suggestion
      <status>tested_robot</status>
```

File: white_balancer/package.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  <author email="info@bit-bots.de">Hamburg Bit-Bots</author>
+
+  <license>MIT</license>
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <depend>image_transport</depend>
+  <depend>roscpp</depend>
+  <depend>sensor_msgs</depend>
+  <depend>std_msgs</depend>
+  <depend>OpenCV</depend>
```
Comment (timonegk) [inline white_balancer/package.xml]: Please add cv bridge dependency

File: white_balancer/src/white_balancer.cpp [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    dynamic_reconfigure::Server<white_balancer::WhiteBalanceConfig>::CallbackType f;
+    f = boost::bind(&WhiteBalancer::callbackRC, this, _1, _2);
+    server.setCallback(f);
+
+    // Register image messages
+    ros::NodeHandle nh;
+    image_transport::ImageTransport it(nh);
+
+    WhiteBalancer::pub = it.advertise("/image", 1);
+    image_transport::Subscriber sub = it.subscribe("/image_unbalanced", 1, &WhiteBalancer::imageCallback, this);
```
Comment (NFiedler) [inline white_balancer/src/white_balancer.cpp]: Why are the topics not configurable? I don't care that much about the dynamic stuff, but it should be at least a YAML parameter. We do not want to recompile the whole thing for a different topic name!

File: white_balancer/src/white_balancer.cpp [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        cv::multiply(cv_ptr->image, cv::Scalar( (float)(255.0/white_value[2]),
+                                                (float)(255.0/white_value[1]),
+                                                (float)(255.0/white_value[0])), cv_ptr->image);
+
+        // Publish white balanced image
+        WhiteBalancer::pub.publish(cv_ptr->toImageMsg());
+    }
+    catch (cv_bridge::Exception& e)
+    {
+        ROS_ERROR("Could not convert from '%s' to 'bgr8'.", msg->encoding.c_str());
```
Comment (NFiedler) [inline white_balancer/src/white_balancer.cpp]: This error message could be throttled.

File: white_balancer/launch/white_balance.launch [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<launch>
+    <node pkg="white_balancer" type="white_balancer" name="white_balancer" output="screen">
+        <remap from="/image_unbalanced" to="/image_raw" />
```
Comment (NFiedler) [inline white_balancer/launch/white_balance.launch]: this is a bit weird

File: white_balancer/cfg/white_balancer.cfg [hunk 8]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python
+# configuration
+
+from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, int_t
+
+gen = ParameterGenerator()
+
+gen.add("temp", int_t, 1, "Color temp for white balance.", min=1000, max=12000, default=4000)
```
Comment (NFiedler) [inline white_balancer/cfg/white_balancer.cfg]: Why is a default set here?

**Pull request merged**